front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -502,5 +521,63 @@ export async function* runRetrieval(
     }
   }
 
-  // Documents are set with a dummy ModelId, now we need to create the actual objects in DB.
+  // We are done, store documents and chunks in database and yield the final events.
+
+  await front_sequelize.transaction(async (t) => {
+    for (const d of documents) {
+      const document = await RetrievalDocument.create(
+        {
+          dataSourceId: d.dataSourceId,
+          sourceUrl: d.sourceUrl,
+          documentId: d.documentId,
+          reference: d.reference,
+          timestamp: d.timestamp,
+          tags: d.tags,
+          score: d.score,
+          retrievalActionId: action.id,
+        },
+        { transaction: t }
+      );
+
+      d.id = document.id;
```

Comment:
NIT: not needed?

Reply:
It is, it's set to 0 above, now it has to be set to the new ModelId.